src/service/ampdoc-impl.js

```diff
@@ -223,6 +285,30 @@ export class AmpDocSingle extends AmpDoc {
   getUrl() {
     return this.win.location.href;
   }
+
+  /** @override */
+  getBody() {
+    return this.win.document.body;
+  }
+
+  /** @override */
+  onBody(callback) {
+    if (this.win.document.body) {
+      callback(this.win.document.body);
```

Comment:
sgtm. it was just a nit we can change this later. you are right `onDocumentReady` already behave this way.